src/components/templates/Audience/Audience.tsx

```diff
@@ -292,21 +292,24 @@ export const Audience: React.FunctionComponent = () => {
     [carvedOutWidthInSeats, carvedOutHeightInSeats]
   );
 
-  // @debt this return useMemo antipattern should be rewritten
-  return useMemo(() => {
-    const takeSeat = (row: number | null, column: number | null) => {
+  const takeSeat = useCallback(
+    (row: number | null, column: number | null) => {
       if (!venueId || !userId) return;
 
       makeUpdateUserGridLocation({
         venueId,
         userUid: userId,
       })(row, column);
-    };
+    },
+    [venueId, userId]
+  );
 
-    const leaveSeat = () => {
-      takeSeat(null, null);
-    };
+  const leaveSeat = useCallback(() => {
+    takeSeat(null, null);
```

Comment:
I guess we use null here because the implementation with undefined is not merged?

Reply:
Where is the implementation with `undefined`?

I didn't seek to change any of the existing functionality in this PR, so I left it as what it was.